testing/unittests/jlibtests.cpp

```diff
@@ -617,4 +619,628 @@ CPPUNIT_TEST_SUITE_REGISTRATION( JlibQuantileTest );
 CPPUNIT_TEST_SUITE_NAMED_REGISTRATION( JlibQuantileTest, "JlibQuantileTest" );
 
 
+//---------------------------------------------------------------------------------------------------------------------
+
+/*
+For comparison, this example of a lock free queue is taken from http://www.1024cores.net/home/lock-free-algorithms/queues/bounded-mpmc-queue
+
+Copyright (c) 2010-2011 Dmitry Vyukov. All rights reserved.
+
+Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
+
+   1. Redistributions of source code must retain the above copyright notice, this list of
+      conditions and the following disclaimer.
+
+   2. Redistributions in binary form must reproduce the above copyright notice, this list
```

Comment:
Are we including this in the compiled code - if so we need to make sure we meet the above condition. If not then worth adding a comment about why we are not. I don't mind leaving the code here, unused, if it is still serving some purpose.